interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala

```diff
@@ -30,53 +30,72 @@ abstract class CatsPlatform extends CatsInstances {
 }
 
 abstract class CatsInstances extends CatsInstances1 {
-  implicit def ioContextShift[E]: ContextShift[IO[E, ?]] = new ContextShift[IO[E, ?]] {
-    override def shift: IO[E, Unit] =
-      IO.yieldNow
+  implicit def ioContextShift[R]: ContextShift[TaskR[R, ?]] = new ContextShift[TaskR[R, ?]] {
+    override def shift: TaskR[R, Unit] =
+      ZIO.yieldNow
 
-    override def evalOn[A](ec: ExecutionContext)(fa: IO[E, A]): IO[E, A] =
+    override def evalOn[A](ec: ExecutionContext)(fa: TaskR[R, A]): TaskR[R, A] =
       fa.on(ec)
   }
 
-  implicit def ioTimer[R <: Clock, E]: effect.Timer[ZIO[R, E, ?]] = new effect.Timer[ZIO[R, E, ?]] {
-    override def clock: cats.effect.Clock[ZIO[R, E, ?]] = new effect.Clock[ZIO[R, E, ?]] {
-      override def monotonic(unit: TimeUnit): ZIO[R, E, Long] =
+  implicit def ioTimer[R <: Clock]: effect.Timer[TaskR[R, ?]] = new effect.Timer[TaskR[R, ?]] {
+    override def clock: cats.effect.Clock[TaskR[R, ?]] = new effect.Clock[TaskR[R, ?]] {
+      override def monotonic(unit: TimeUnit): TaskR[R, Long] =
         zioClock.nanoTime.map(unit.convert(_, NANOSECONDS))
 
-      override def realTime(unit: TimeUnit): ZIO[R, E, Long] =
+      override def realTime(unit: TimeUnit): TaskR[R, Long] =
         zioClock.currentTime(unit)
     }
 
-    override def sleep(duration: FiniteDuration): ZIO[R, E, Unit] =
+    override def sleep(duration: FiniteDuration): TaskR[R, Unit] =
       zioClock.sleep(scalaz.zio.duration.Duration.fromNanos(duration.toNanos))
   }
 
-  implicit val taskEffectInstances: effect.ConcurrentEffect[Task] with SemigroupK[Task] =
-    new CatsConcurrentEffect
+  implicit def taskEffectInstances(implicit runtime: Runtime[Any]): effect.ConcurrentEffect[Task] =
+    new CatsConcurrentEffect { override val rts = runtime }
 
-  implicit val taskParallelInstance: Parallel[Task, Util.Par] =
+  implicit def taskConcurrentInstances[R]: effect.Concurrent[TaskR[R, ?]] =
+    new CatsConcurrent[R]
+
+  implicit def taskParallelInstance(implicit rts: Runtime[Any]): Parallel[Task, Util.Par] =
```

Comment:
Change to `Runtime[R]` and generalize `Task` to `TaskR`.

Reply:
Actually there's another parallel instance down below that will clash with this one, just delete this one.